server/src/services/sologame.ts

```diff
@@ -0,0 +1,202 @@
+import { BaseService } from "@port-of-mars/server/services/db";
+import { EventCardData, SoloGameStatus, TreatmentData } from "@port-of-mars/shared/sologame";
+import {
+  SoloGame,
+  SoloGameRound,
+  SoloGameTreatment,
+  SoloMarsEventCard,
+  SoloMarsEventDeck,
+  SoloMarsEventDeckCard,
+  SoloPlayer,
+  SoloPlayerDecision,
+  User,
+} from "@port-of-mars/server/entity";
+import { getRandomIntInclusive } from "@port-of-mars/server/util";
+import { SoloGameState } from "@port-of-mars/server/rooms/sologame/state";
+
+export class SoloGameService extends BaseService {
+  async drawEventCardDeck(): Promise<EventCardData[]> {
+    /**
+     * draw a deck of event cards from the db
+     */
+    const cards = await this.em.getRepository(SoloMarsEventCard).find();
+    const deck: EventCardData[] = [];
+
+    for (const card of cards) {
+      const drawAmt = getRandomIntInclusive(card.drawMin, card.drawMax);
+      for (let i = 0; i < drawAmt; i++) {
+        // card effects are encoded in the db with a range of possible rolls and
+        // a multiplier for each value (sys health, points, .. ), this is typically
+        // 0, 1 or -1
+        const roll = getRandomIntInclusive(card.rollMin, card.rollMax);
+        // fill out templates from the db with the actual roll value and pluralization
+        const effectText = card.effect
+          .replace("{roll}", roll.toString())
```

Comment:
these dynamics template strings could also be set up as functions in the EventCards themselves, something like https://stackoverflow.com/a/40147828/93370

(for the future)